src/Build/Utilities/EngineFileUtilities.cs

```diff
@@ -154,11 +191,109 @@ FileMatcher fileMatcher
                 // as a relative path, we will get back a bunch of relative paths.
                 // If the filespec started out as an absolute path, we will get
                 // back a bunch of absolute paths.
-                fileList = fileMatcher.GetFiles(directoryUnescaped, filespecUnescaped, excludeSpecsUnescaped);
+                (fileList, action, excludeFileSpec) = fileMatcher.GetFiles(directoryUnescaped, filespecUnescaped, excludeSpecsUnescaped);
+                if (action == FileMatcher.SearchAction.LogDriveEnumeratingWildcard)
+                {
+                    switch (loggingMechanism)
+                    {
+                        case TargetLoggingContext targetLoggingContext:  // Logging mechanism used for ItemGroupIntrinsicTask
+                            if (((!string.IsNullOrWhiteSpace(excludeFileSpec)) && (!disableExcludeDriveEnumerationWarning)) || (includeLocation == null)) // First condition necessary for 1st GetFileListEscaped call to not get caught here, second condition necessary for GetFileListUnescaped call to get caught here.
```

Comment:
Some of these lines are shockingly long. Would you mind reformatting them? We don't have a length limit in the MSBuild codebase but something like 120 is probably best.